tapa/verilog/xilinx/module.py

```diff
@@ -831,6 +831,29 @@ def get_streams_fifos(module: Module, streams_name: str) -> list[str]:
             number = match.group(1)
             fifos.add(f"{streams_name}_{number}")
 
+    # singleton array without number
+    # when a stream array has length 1,
+    # the generated port name may not have the number
+    # in it.
+    # e.g., in tests/functional/singleton/vadd.cpp,
+    # "tapa::istreams<float, M>& a" where M = 1 resulted in
+    # stream data port a_s_dout rather than a_0_dout.
+    if not fifos:
+        infixes = ("_V", "_r", "_s", "")  # vitis hls generated port infixes
```

Comment:
Please move this list to a constant and reference the same from here and from `get_port_of` to make it easier to update this list in the future.